src/Containers/InjectedNavigationContainer.tsx

```diff
@@ -42,7 +41,6 @@ export const InjectedNavigationContainer = ({
   onStateChange,
 }: Props): JSX.Element => {
   useFlipper(navigationRef)
-  useAndroidBackNavigation(navigationRef)
```

Comment:
why this original logic was not working, what is the difference ?

Reply:
The original logic was doing a back navigation to SplashScreen, which then exited the app. In new logic, we prevent this by exiting the app on the initial screen. This hook now needs to be called in the `withScreen`.

Reply:
let's use this one and extend it with detection for is initial.